### PR TITLE
add `cupy.random.logseries`

### PR DESCRIPTION
**Pull request by YoshikawaMasashi**

### CODE REVIEW

**Comment by YoshikawaMasashi** [comment]
Jenkins, test this please.

File: cupy/random/generator.py [hunk 1]
```diff
@@ -201,6 +201,55 @@ def lognormal(self, mean=0.0, sigma=1.0, size=None, dtype=float):
...
+        while (1) {
+            if (V >= p) {
+                y = 1;
+                break;
+            }
+            q = 1.0 - exp(r*U);
+            if (V <= q*q) {
+                result = (T)floor(1 + log(V)/log(q));
+                if (result < 1) {
+                    continue;
```
Comment (toslunar) [inline cupy/random/generator.py]: On this case, `continue` should regenerate the uniform randoms.

**Comment by YoshikawaMasashi** [comment]
Jenkins, test this please.

**Comment by toslunar** [comment]
Could you add a test to check the range of `p`?
```
>>> np.random.logseries(0)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "mtrand.pyx", line 4365, in mtrand.RandomState.logseries
ValueError: p <= 0.0
>>> np.random.logseries(1)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "mtrand.pyx", line 4367, in mtrand.RandomState.logseries
ValueError: p >= 1.0
```

**Comment by YoshikawaMasashi** [comment]
Jenkins, test this please.

File: tests/cupy_tests/random_tests/test_distributions.py [hunk 2]
```diff
@@ -256,6 +256,18 @@ def test_logseries(self, p_dtype, dtype):
...
+    @cupy.testing.for_dtypes([numpy.int64, numpy.int32], 'dtype')
+    @cupy.testing.for_float_dtypes('p_dtype', no_float16=True)
+    def test_logseries_for_invalid_p(self, p_dtype, dtype):
+        with self.assertRaises(ValueError):
+            cp_params = {'p': cupy.zeros(self.p_shape, dtype=p_dtype)}
+            getattr(distributions, 'logseries')(
+                size=self.shape, dtype=dtype, **cp_params)
+        with self.assertRaises(ValueError):
+            cp_params = {'p': cupy.ones(self.p_shape, dtype=p_dtype)}
+            getattr(distributions, 'logseries')(
```
Comment (toslunar) [inline tests/cupy_tests/random_tests/test_distributions.py]: Why is `getattr` needed?

**Comment by YoshikawaMasashi** [comment]
Jenkins, test this please.

**Comment by toslunar** [comment]
Jenkins, test this please.

**Comment by toslunar** [comment]
Jenkins, test this please.

**Pull request merged**